File: app/src/main/java/com/king/app/coolg_kt/model/http/AppService.java
package com.king.app.coolg_kt.model.http;

import com.king.app.coolg_kt.model.http.bean.request.FolderRequest;
import com.king.app.coolg_kt.model.http.bean.request.GdbCheckNewFileBean;
import com.king.app.coolg_kt.model.http.bean.request.GdbRequestMoveBean;
import com.king.app.coolg_kt.model.http.bean.request.GetStarRatingsRequest;
import com.king.app.coolg_kt.model.http.bean.request.PathRequest;
import com.king.app.coolg_kt.model.http.bean.request.UploadStarRatingRequest;
import com.king.app.coolg_kt.model.http.bean.request.VersionRequest;
import com.king.app.coolg_kt.model.http.bean.response.AppCheckBean;
import com.king.app.coolg_kt.model.http.bean.response.BaseResponse;
import com.king.app.coolg_kt.model.http.bean.response.FolderResponse;
import com.king.app.coolg_kt.model.http.bean.response.GdbMoveResponse;
import com.king.app.coolg_kt.model.http.bean.response.GdbRespBean;
import com.king.app.coolg_kt.model.http.bean.response.GetStarRatingResponse;
import com.king.app.coolg_kt.model.http.bean.response.OpenFileResponse;
import com.king.app.coolg_kt.model.http.bean.response.PathResponse;
import com.king.app.coolg_kt.model.http.bean.response.VersionResponse;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * Created by Administrator on 2016/9/5.
 */
public interface AppService {
    @GET("online")
    Observable<GdbRespBean> isServerOnline();

    @GET("checkNew")
    Observable<AppCheckBean> checkAppUpdate(@Query("type") String type, @Query("version") String version);

    @GET("checkNew")
    Observable<AppCheckBean> checkGdbDatabaseUpdate(@Query("type") String type, @Query("version") String version);

    @GET("checkNew")
    Observable<GdbCheckNewFileBean> checkNewFile(@Query("type") String type);

    @POST("requestMove")
    Observable<GdbMoveResponse> requestMoveImages(@Body GdbRequestMoveBean data);

    @POST("surfFolder")
    Observable<FolderResponse> requestSurf(@Body FolderRequest data);

    @POST("uploadStarRatings")
    Observable<BaseResponse> uploadStarRatings(@Body UploadStarRatingRequest data);

    @POST("getStarRatings")
    Observable<BaseResponse<GetStarRatingResponse>> getStarRatings(@Body GetStarRatingsRequest data);

    @POST("videoPath")
    Observable<PathResponse> getVideoPath(@Body PathRequest data);

    @POST("openFile")
    Observable<OpenFileResponse> openFileOnServer(@Body PathRequest data);

    @POST("version")
    Observable<BaseResponse<VersionResponse>> getVersion(@Body VersionRequest data);
}

File: app/src/main/java/com/king/app/coolg_kt/model/http/AppService.kt
package com.king.app.coolg_kt.model.http

import com.king.app.coolg_kt.model.http.bean.request.*
import com.king.app.coolg_kt.model.http.bean.response.*
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

/**
 * Created by Administrator on 2016/9/5.
 */
interface AppService {

    @GET("online")
    fun isServerOnline(): Observable<GdbRespBean>

    @GET("checkNew")
    fun checkAppUpdate(
        @Query("type") type: String,
        @Query("version") version: String
    ): Observable<AppCheckBean>

    @GET("checkNew")
    fun checkGdbDatabaseUpdate(
        @Query("type") type: String,
        @Query("version") version: String
    ): Observable<AppCheckBean>

    @GET("checkNew")
    fun checkNewFile(@Query("type") type: String): Observable<GdbCheckNewFileBean>

    @POST("requestMove")
    fun requestMoveImages(@Body data: GdbRequestMoveBean): Observable<GdbMoveResponse>

    @POST("surfFolder")
    fun requestSurf(@Body data: FolderRequest): Observable<FolderResponse>

    @POST("subtitle")
    fun searchSubtitle(@Body data: SubtitleRequest): Observable<SubtitleResponse>

    @POST("uploadStarRatings")
    fun uploadStarRatings(@Body data: UploadStarRatingRequest): Observable<BaseResponse<*>>

    @POST("getStarRatings")
    fun getStarRatings(@Body data: GetStarRatingsRequest): Observable<BaseResponse<GetStarRatingResponse>>

    @POST("videoPath")
    fun getVideoPath(@Body data: PathRequest): Observable<PathResponse>

    @POST("openFile")
    fun openFileOnServer(@Body data: PathRequest): Observable<OpenFileResponse>

    @POST("version")
    fun getVersion(@Body data: VersionRequest): Observable<BaseResponse<VersionResponse>>
}